x-pack/platform/plugins/shared/fleet/public/applications/integrations/app.tsx

```diff
@@ -141,6 +144,29 @@ export const IntegrationsAppContext: React.FC<{
 export const AppRoutes = memo(() => {
   const flyoutContext = useFlyoutContext();
   const fleetStatus = useFleetStatus();
+  const authz = useAuthz();
+  const isAddIntegrationsPath = !!useRouteMatch(
+    INTEGRATIONS_ROUTING_PATHS.add_integration_to_policy
+  );
+  const allowedToAccess = authz.integrations.readIntegrationPolicies || authz.fleet.all;
+  const [permissionsErrors, setPermissionsErrors] = useState<string>();
+
+  useEffect(() => {
```

Comment:
you could replace `useEffect` and `useState` with `useMemo`, something like this:
```
const permissionsError = useMemo(() => allowedToAccess ? 'MISSING_PRIVILEGES' : undefined, [allowedToAccess]);
```

Reply:
I would go so far as to say `useMemo` isn't adding a meaningful performance improvement here and may as well just simplify and calculate this on each render. It's a single boolean check.